app/src/common/shared/org/mozilla/vrbrowser/ui/views/AutoCompletionView.java

```diff
@@ -60,6 +62,8 @@ public AutoCompletionView(Context aContext, AttributeSet aAttrs, int aDefStyle)
     private void initialize(Context aContext) {
         inflate(aContext, R.layout.autocompletion_bar, this);
         mFirstLine = findViewById(R.id.autoCompletionFirstLine);
+        mLineWidth = WidgetPlacement.pixelDimension(getContext(), R.dimen.world_width);
```

Comment:
It's not correct to use the world width here. Ideally the View should get the current view width and detect layout changes. You can override `onSizeChanged` in order to detect the size and layout of the whole view: https://developer.android.com/reference/android/view/View.html#onSizeChanged%28int,%20int,%20int,%20int%29

Reply:
@MortimerGoro I think it is good enough after adding
```
@Override
    public void onSizeChanged(int w, int h, int oldw, int oldh) {
        if (mLineWidth == 0) {
            View scrollView = findViewById(R.id.firstLineScroll);
            scrollView.measure(MeasureSpec.UNSPECIFIED, MeasureSpec.UNSPECIFIED);
            mLineWidth = scrollView.getWidth();
        }
    }
```

The `onSizeChanged` will be delay dispatched after the first time of `setItems()`. Therefore, the mLineWidth is still zero, we need to use kMaxItemsPerLine to workaround this.